nav2_amcl/src/amcl_node.cpp

```diff
@@ -1053,31 +1051,12 @@ AmclNode::handleMapMessage(const nav_msgs::msg::OccupancyGrid & msg)
       global_frame_id_.c_str());
   }
 
-  freeMapDependentMemory();
```

Comment:
I'm not completely familiar with this code, so forgive me if I'm misunderstanding. But now it seems you are never calling this function `freeMapDependentMemory()` anywhere, and are thus never freeing map or clearing laser objects. Is that intentional here?

Reply:
good catch. This got deleted by accident.